app/services/EventReportService.scala

```diff
@@ -44,28 +46,22 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
   private val compileEventOneReportSchemaPath = "/resources.schemas/api-1827-create-compiled-event-1-report-request-schema-v1.0.1.json"
   private val compileMemberEventReportSchemaPath = "/resources.schemas/api-1830-create-compiled-member-event-report-request-schema-v1.0.4.json"
 
-
-  def compileEventReport(pstr: String, userAnswersJson: JsValue)
+  def compileEventReport(pstr: String, eventType: EventType)
                         (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
 
-    val maybeApi1826 = eventReportCacheRepository.getByKeys(Map("pstr" -> pstr, "apiTypes" -> Api1826.toString)).map {
-      case Some(data) => compileEventReportSummary(pstr, data).map(_ => NoContent)
-      case _ => Future.successful(Ok)
-    }.flatten
-
-    val maybeApi1827 = eventReportCacheRepository.getByKeys(Map("pstr" -> pstr, "apiTypes" -> Api1827.toString)).map {
-      case Some(data) => compileEventOneReport(pstr, data).map(_ => NoContent)
-      case _ => Future.successful(Ok)
-    }.flatten
-
-    val maybeApi1830 = eventReportCacheRepository.getByKeys(Map("pstr" -> pstr, "apiTypes" -> Api1830.toString)).map {
-      case Some(data) => compileMemberEventReport(pstr, data).map(_ => NoContent)
-      case _ => Future.successful(Ok)
-    }.flatten
-
-    val seqOfMaybeApiCalls = Future.sequence(Seq(maybeApi1826, maybeApi1827, maybeApi1830))
+    val apiType = EventType.postApiTypeByEventType(eventType)
+    val performCompile: (String, JsValue) => Future[Result] =
+      apiType match {
+        case Api1826 => compileEventReportSummary _
+        case Api1827 => compileEventOneReport _
+        case Api1830 => compileMemberEventReport _
+        case _ => (_, _) => Future.successful(NoContent)
+      }
```

Comment:
we are always returning Future.successful(NoContent) for every case even it's not really successful. Wouldn't it be better if you use different HTTP codes ?

Reply:
Well, I did think about this. But we don't at present have a separate "compile failed" page in the frontend. If anything fails, e.g. schema validation then an exception should be thrown. Perhaps we should throw an exception in the above case too, i.e. we have no compile implementation for an event type.